src/File.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace IterTools;
+
+use IterTools\Util\ResourceHelper;
+
+class File
+{
+    /**
+     * Reads file resource line by line.
+     *
+     * @param resource $file
+     *
+     * @return \Generator<string>
+     *
+     * @throws \UnexpectedValueException if invalid resource given
+     *
+     * @see fgets()
+     */
+    public static function readLines($file): \Generator
+    {
+        ResourceHelper::checkIsValid($file);
+
+        while (($line = \fgets($file)) !== false) {
+            yield $line;
+            ResourceHelper::checkIsValid($file);
```

Comment:
Question: Why is this check required after each line as it was already determined to be a valid resource to get into the while loop?

Reply:
I agree with you. This is definitely an unnecessary check. Removed it.